zerver/actions/user_settings.py

```diff
@@ -152,15 +152,33 @@ def do_start_email_change_process(user_profile: UserProfile, new_email: str) ->
         activate_url=activation_url,
     )
     language = user_profile.default_language
-    send_email(
-        "zerver/emails/confirm_new_email",
-        to_emails=[new_email],
-        from_name=FromAddress.security_email_from_name(language=language),
-        from_address=FromAddress.tokenized_no_reply_address(),
-        language=language,
-        context=context,
-        realm=user_profile.realm,
-    )
+
+    if old_email == "":
+        # The assertions here are to help document the only circumstance under which
+        # this condition should be possible.
+        assert (
+            user_profile.realm.demo_organization_scheduled_deletion_date is not None
+            and user_profile.is_realm_owner
+        )
+        send_email(
+            "zerver/emails/confirm_demo_organization_email",
+            to_emails=[new_email],
+            from_name=FromAddress.security_email_from_name(language=language),
+            from_address=FromAddress.tokenized_no_reply_address(),
+            language=language,
+            context=context,
+            realm=user_profile.realm,
+        )
+    else:
+        send_email(
+            "zerver/emails/confirm_new_email",
+            to_emails=[new_email],
+            from_name=FromAddress.security_email_from_name(language=language),
+            from_address=FromAddress.tokenized_no_reply_address(),
+            language=language,
+            context=context,
+            realm=user_profile.realm,
+        )
```

Comment:
If the only thing different in this code path is the template name, we can probably make that a variable and be the only thing inside the `if/else` block?